### PR TITLE
Combine several tasks into single task and added retry for network related tasks

### PR DESCRIPTION
**Pull request by dj-wasabi**
This PR contains the following change:

* Combined similar tasks into a single task;
* Added retries for tasks that does something with network related services, to prevent single network issues (Ansible best practice);
* Only run some tasks when other task is not changed;

### CODE REVIEW

File: roles/php/tasks/main.yml [hunk 1]
```diff
@@ -81,10 +83,10 @@
...
     - "restart php72-php-fpm"
 
-- name: Enable php72-php-fpm
-  service:
-    name: php72-php-fpm
-    enabled: yes
+# - name: Enable php72-php-fpm
+#   service:
+#     name: php72-php-fpm
+#     enabled: yes
```
Comment (quartje) [inline roles/php/tasks/main.yml:90]: Why is this task commented?

**Review by quartje — COMMENTED** [review]
Nice work! That seems to simplify a lot. 
Could you explain why the 'untils' are added? Is that Ansible best practice behaviour?

File: roles/php/tasks/main.yml (hunk 1 above)
Comment (dj-wasabi) [reply roles/php/tasks/main.yml:90]: I commented it (Would actually wanted to remove it) but the handler is already enabling and (re)starting the service. So having a 2nd task which only does the enabling is a bit double.

Comment (quartje) [reply roles/php/tasks/main.yml:90]: Right, removing it should be ok than!

**Comment by dj-wasabi** [comment]
> Nice work! That seems to simplify a lot.
> Could you explain why the 'untils' are added? Is that Ansible best practice behaviour?

Yes, it is considered to be a best practice. Assume that network related tasks could experience a timeout, hickup or any other temporary failure. With adding the `until` option, the task will be executed 3 times (Unless you specify more) before it actually fails and thus Ansible will try for example 3 times to install an package from the internet before it fails.

File: roles/php/tasks/main.yml (hunk 1 above)
Comment (dj-wasabi) [reply roles/php/tasks/main.yml]: Yes, removed it. Thanks! 👍

**Comment by quartje** [comment]
> > Nice work! That seems to simplify a lot.
> > Could you explain why the 'untils' are added? Is that Ansible best practice behaviour?
> 
> Yes, it is considered to be a best practice. Assume that network related tasks could experience a timeout, hickup or any other temporary failure. With adding the `until` option, the task will be executed 3 times (Unless you specify more) before it actually fails and thus Ansible will try for example 3 times to install an package from the internet before it fails.

:+1:

**Pull request merged**